walk-in-interview/src/main/java/com/google/job/data/JobStatus.java

```diff
@@ -0,0 +1,34 @@
+package com.google.job.data;
+
+/** Enumeration that represents the status of a job post. */
+public enum JobStatus {
+    ACTIVE("ACTIVE"),
+    DELETED("DELETED"),
+    EXPIRED("EXPIRED");
+
+    private final String statusId;
+
+    JobStatus(String statusId) {
+        this.statusId = statusId;
+    }
+
+    /**
+     * Gets the id string.
+     *
+     * @return Id string.
+     */
+    public String getStatusId() {
+        return statusId;
+    }
+
+    /**
+     * Returns the job status enum matching the provided id.
+     *
+     * @param id String id matching the enum value.
+     * @return Status enum matching the provided id.
+     * @throws IllegalArgumentException If a status cannot be found for provided id.
+     */
+    public static JobStatus getFromId(String id) throws IllegalArgumentException {
+        return JobStatus.valueOf(id);
```

Comment:
this is incorrect. the id matches the statusId not the enum name:
```
for( JobStatus status : values() ){
   if( status.getStatusId().equals(id) ){
       return status;
    }
}
throw new IllegalArgumentException("Invalid job status id: "+id);
```

Reply:
Fixed